### PR TITLE
feat: improve dashboard filter layout

### PR DESCRIPTION
**Pull request by magnew**
Closes: #6477 

### Description:

Improve the dashboard filter interaction. Mainly this combines the FilterSearch and FilterConfiguration components and moves certain elements under the tabs.

### CODE REVIEW

File: packages/frontend/src/components/DashboardFilter/index.tsx [hunk 1]
```diff
@@ -1,31 +1,72 @@
...
 
 const DashboardFilter: FC<Props> = ({ isEditMode }) => {
+    const { track } = useTracking();
     const { projectUuid } = useParams<{ projectUuid: string }>();
-    const [isFilterPopoverOpen, setIsFilterPopoverOpen] = useState(false);
+    const [selectedField, setSelectedField] = useState<FilterableField>();
+    const [isPopoverOpen, { close: closePopover, toggle: togglePopover }] =
+        useDisclosure();
+    const [isSubPopoverOpen, { close: closeSubPopover, open: openSubPopover }] =
+        useDisclosure();
```
Comment (joaoviana) [inline packages/frontend/src/components/DashboardFilter/index.tsx:33]: 🥇

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx [hunk 2]
```diff
@@ -248,12 +294,14 @@ const FilterConfiguration: FC<Props> = ({
                             internalFilterRule,
                         )
                     }
-                    onClick={() => onSave(internalFilterRule)}
+                    onClick={() => {
+                        if (!!internalFilterRule) onSave(internalFilterRule);
```
Comment (joaoviana) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:298]: here you can add the `setSelectedTabId(FilterTabs.SETTINGS)` 👍

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx [hunk 3]
```diff
@@ -161,64 +173,98 @@ const FilterConfiguration: FC<Props> = ({
...
-                        popoverProps={popoverProps}
-                    />
-                </Tabs.Panel>
+                <Tabs.Panel value={FilterTabs.SETTINGS} w={350}>
+                    <Stack>
+                        {!!fields && isCreatingNew ? (
+                            <FormGroup
+                                style={{ marginBottom: '5px' }}
+                                label={
+                                    <Text>Select a dimension to filter</Text>
```
Comment (joaoviana) [inline packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:212]: from the designs i would add: 
```suggestion
                                    <Text fw={500} >Select a dimension to filter</Text>
```

and a start to indicate it's required... perhaps `useForm` from `mantine`? 🤔

**Review by joaoviana — CHANGES_REQUESTED** [review]
looks good! And great refactoring on `FilterSearch` vs what we can currently re-use 🎉 

Small UI change 👍

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx (hunk 2 above)
Comment (magnew) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:299]: Yup, done in rebase.

File: packages/frontend/src/components/DashboardFilter/index.tsx [hunk 4]
```diff
@@ -74,26 +117,35 @@ const DashboardFilter: FC<Props> = ({ isEditMode }) => {
...
-                            fields={allFilterableFields || []}
-                            popoverProps={{
-                                usePortal: false,
-                            }}
-                            onClose={() => setIsFilterPopoverOpen(false)}
-                        />
+                    <Popover.Dropdown ml={5}>
+                        {filterableFieldsByTileUuid ? (
+                            <FilterConfiguration
+                                isCreatingNew
```
[the remark this answers is not in the document]
Comment (magnew) [reply packages/frontend/src/components/DashboardFilter/index.tsx:131]: Yea, this add button always treats it as creating new

File: packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx (hunk 3 above)
Comment (magnew) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:212]: I will add the fw and star, but not bring in useForm for now. There's a lot to that. The mantine input has 'required' 😢

Comment (joaoviana) [reply packages/frontend/src/components/DashboardFilter/FilterConfiguration/index.tsx:212]: sounds good!

**Review by joaoviana — APPROVED** [review]

**Pull request merged**